### PR TITLE
Add PlatformSpec for buildfile

### PR DESCRIPTION
**Pull request by loosebazooka**
part of #2570

### CODE REVIEW

File: jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+package com.google.cloud.tools.jib.cli.buildfile;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/**
+ * A yaml block for specifying platforms
```
Comment (chanseokoh) [inline jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java:26]:
```suggestion
 * A yaml block for specifying platforms.
```

File: jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+   * @param osFeatures a list of operating system features
+   */
+  @JsonCreator
+  public PlatformSpec(
+      @JsonProperty("architecture") String architecture,
+      @JsonProperty("os") String os,
+      @JsonProperty("os.version") String osVersion,
+      @JsonProperty("os.features") List<String> osFeatures,
+      @JsonProperty("variant") String variant,
+      @JsonProperty("features") List<String> features) {
```
Comment (chanseokoh) [inline jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java:67]: For every class in jib-build-plan, I copied collections to make it truly immutable. Should it do the same?

File: jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  public Optional<String> getOs() {
+    return Optional.ofNullable(os);
+  }
+
+  public Optional<String> getOsVersion() {
+    return Optional.ofNullable(osVersion);
+  }
+
+  public Optional<List<String>> getOsFeatures() {
+    return Optional.ofNullable(osFeatures);
```
Comment (chanseokoh) [inline jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java:89]: Likewise, it's an overhead, but every class in jib-build-plan returns a copy.

Comment (loosebazooka) [reply jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java:89]: Don't think users will be interacting with this code. Perhaps I could skip `Optional` as well? I see how it flows when I'm building the buildplan.

Comment (chanseokoh) [reply jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java:89]: Honestly, I am still unsure on this `Optional` topic and would love to get an expert guide. Generally speaking, getting rid of null (and null checks) by using `Optional` to denote something (especially for public API) is absent sounds like a good idea. However, sometimes it results in more bloated code that is really difficult to tolerate. And I'm also unclear if there should be a difference when using `Optional` as a return value versus as method arguments.

File: jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java (hunk 2 above)
Comment (chanseokoh) [reply jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java:67]: Saw the comment in the other thread. Marking resolved.

**Pull request merged**